src/vs/vscode.d.ts

```diff
@@ -9024,57 +9024,61 @@ declare module 'vscode' {
 		dispose(): void;
 	}
 
+	/**
+	 * Author information of a [comment](#Comment)
+	 */
+
+	export interface CommentAuthorInformation {
+		/**
+		 * The display name of the author of the comment
+		 */
+		name: string;
+
+		/**
+		 * The optional icon path for the author
+		 */
+		iconPath?: Uri;
+	}
+
 	/**
 	 * A comment is displayed within the editor or the Comments Panel, depending on how it is provided.
 	 */
-	export class Comment {
+	export interface Comment {
 		/**
 		 * The id of the comment
 		 */
-		readonly id: string;
+		id: string;
```

Comment:
I think it still makes sense to mark properties that aren't updateable as `readonly`. Seems like `id` and `author` shouldn't change

Reply:
The reason I don't use any `readonly` here is because we don't do any validation or protection on them, `Comment` is copied to the renderer side once `CommentThread.comments` is set. It also means that adding `readonly` to these properties only change how extensions authors program but don't add any more type safe as they can have their own definition of `Comment` with the same shape but without any `readonly` accessor. So I think we can give the freedom to extension.

Reply:
Ok, sounds good!